src/gui_common/charts/line/LineChart.cs

```diff
@@ -299,11 +319,16 @@ public void ClearDataSets()
                 visibleDataSetCount++;
             }
 
-            foreach (var point in data.Value.DataPoints)
+            // Initialize line
+            if (!string.IsNullOrEmpty(data.Key))
```

Comment:
would it make sense to throw an exception here? or would the code actually gracefully handle the condition that data.Key is null?